crates/core/src/host/module_host.rs

```diff
@@ -636,12 +637,19 @@ impl ModuleHost {
         &self.info.subscriptions
     }
 
-    async fn call<F, R>(&self, _reducer_name: &str, f: F) -> Result<R, NoSuchModule>
+    async fn call<F, R>(&self, reducer: &str, f: F) -> Result<R, NoSuchModule>
     where
         F: FnOnce(&mut dyn ModuleInstance) -> R + Send + 'static,
         R: Send + 'static,
     {
-        let mut inst = self.inner.get_instance(self.info.address).await?;
+        let mut inst = {
+            // Record the time spent waiting in the queue
+            let _guard = WORKER_METRICS
+                .reducer_wait_time
+                .with_label_values(&self.info.address, reducer)
+                .start_timer();
+            self.inner.get_instance(self.info.address).await?
```

Comment:
@coolreader18 do you agree that this is the correct place to be recording this metric? And that this works for all reducers?

Reply:
for all reducers and for other such calls. but yeah, this seems right.